pkg/filesystem/path/resolve.go

```diff
@@ -47,18 +47,14 @@ func (rs *resolverState) resolve() error {
 		case GotDirectory:
 			rs.componentWalker = rv.Child
 		case GotSymlink:
-			target, err := NewUNIXParser(rv.Target)
-			if err != nil {
+			if err := rs.push(rv.Parent, rv.Target); err != nil {
 				return err
 			}
-			if err := rs.push(rv.Parent, target); err != nil {
-				return err
-			}
-		default:
-			panic("Missing result")
+	default:
+		panic("Missing result")
 		case nil: // Do nothing
```

Comment:
I think `case nil` only happens if we've exhausted path resolution, right? Because we get it when calling `OnTerminal()`. Maybe it's worth making that a bit more explicit?
```go
case nil:
    // Parsed a terminal component. This should only be permitted at the very end of the path resolution process.
    if len(rs.stack) > 0 {
        panic("Parser.ParseFirstComponent() did not respect mustBeDirectory")
    }
```